[test_hybrid_data_tensor_parallel_example_T3000.py] | (T3K) T3000 unit tests | t3000-unit-tests / t3k_ttnn_tests [wh_llmbox]

## CI Failure Report

<!-- AUTO-TRIAGE-SUMMARY-START -->
**Summary:** [CI] t3k_ttnn_tests [wh_llmbox] fails with "pytest: error: unrecognized arguments: --count=3" — pytest-repeat plugin missing
<!-- AUTO-TRIAGE-SUMMARY-END -->

**Workflow:** (T3K) T3000 unit tests
**Job:** t3000-unit-tests / t3k_ttnn_tests [wh_llmbox]

### Regression timeline

- **Last passing commit:** N/A (not provided)
- **First failing commit:** `b19100423b70` — 2026-07-08T06:48:16Z — [job link](https://github.com/tenstorrent/tt-metal/actions/runs/28923347893/job/85807010367)
- **Most recent failing commit:** `b5e3bb006104` — 2026-07-08T12:31:29Z — [job link](https://github.com/tenstorrent/tt-metal/actions/runs/28942749986/job/85873927413)

### Failing job URLs (last 2 runs)

- https://github.com/tenstorrent/tt-metal/actions/runs/28942749986/job/85873927413
- https://github.com/tenstorrent/tt-metal/actions/runs/28923347893/job/85807010367

### Failing test path(s)

The failure occurs on the final pytest invocation in the job, which uses `--count=3`. This is the last test command after `tests/ttnn/distributed/test_hybrid_data_tensor_parallel_example_T3000.py` completes (1 skipped). The exact test file/path could not be determined because pytest aborts during argument parsing before collecting tests.

- Likely uses `pytest-repeat` plugin (provides `--count` flag), which is not installed in the environment.

### Error excerpt (terminal failure from logs)

**Run 1** (commit `b5e3bb006104`, ~13:22:34 UTC):
```
ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: --count=3
  inifile: /work/pytest.ini
  rootdir: /work

##[error]Process completed with exit code 4.
```

**Run 2** (commit `b19100423b70`, ~07:54:01 UTC):
```
ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: --count=3
  inifile: /work/pytest.ini
  rootdir: /work

##[error]Process completed with exit code 4.
```

### Reproduction steps

1. Checkout the failing commit (e.g. `b5e3bb006104`).
2. Run the t3k_ttnn_tests job on a wh_llmbox runner.
3. The job will fail on the final pytest invocation that includes `--count=3`.
4. To reproduce the specific failure locally:
   ```bash
   # Confirm pytest-repeat is not installed:
   pip show pytest-repeat
   # Then run the test command that uses --count=3 (check CI scripts for the exact command)
   pytest --count=3 <test_file>
   ```
5. Fix: install the `pytest-repeat` plugin: `pip install pytest-repeat`

### Notes

- Pytest exit code 4 = "command line usage error" — the `--count=N` flag is provided by the `pytest-repeat` plugin. Since the plugin is missing from the installed packages, pytest rejects the argument and aborts before any test collection.
- All prior test sessions in the same job run successfully (passed or skipped). The failure is isolated to the last invocation that uses `--count=3`.
- The `pytest-repeat` plugin must have been removed or was never added to the environment's dependencies around the time of commit `b19100423b70` (2026-07-08T06:48:16Z).

**Determinism check:**
- Run 1 terminal error (line 91927): `pytest: error: unrecognized arguments: --count=3`
- Run 2 terminal error (line 91919): `pytest: error: unrecognized arguments: --count=3`
- Both runs exit with code 4; error messages are word-for-word identical.

<!-- AUTO-TRIAGE-METADATA-START -->
`Auto-triage-workflow: (T3K) T3000 unit tests`
`Auto-triage-job-name: t3000-unit-tests / t3k_ttnn_tests [wh_llmbox]`
<!-- AUTO-TRIAGE-METADATA-END -->

<!-- AUTO-TRIAGE-INFO-START -->
## About this issue (automated CI auto-triage)
This issue was opened and assigned by an automated AI system. It posts through a bot token, so notifications may look like they come from a person, but they are automated.

The owner is selected from recent commit history and job-ownership data. It reflects who is *likely able to fix or route* this, not necessarily who caused it, so it can be imperfect.

**If this was assigned to the wrong person, you can:**
- Comment `REASSIGN OWNER: <why this is outside your area / which team or test-area it belongs to>` and the system will automatically re-route it to a better-suited owner.
- Unassign yourself (this stops notifications to you).
- Close the issue if it is not a real or relevant failure.

The assigned owner is reminded about every 3 days until the issue is resolved. Full guide: https://tenstorrent.atlassian.net/wiki/spaces/MI6/pages/2545221692/Auto-Triage+Info
<!-- AUTO-TRIAGE-INFO-END -->

<!-- PIPELINE-REPORT-START -->
## Pipeline report
Analysis report: https://github.com/tenstorrent/tt-metal/actions/runs/28881020681
Slack notification: https://tenstorrent.slack.com/archives/C0B1F8Z7GQM/p1783441505716629?thread_ts=1783440649.735139&cid=C0B1F8Z7GQM
<!-- PIPELINE-REPORT-END -->

--- a/tt_metal/python_env/requirements-dev.txt
+++ b/tt_metal/python_env/requirements-dev.txt
@@ -42,6 +42,7 @@ pytest-cov==7.0.0
 pytest-timeout==2.4.0
 pytest-split==0.11.0
 pytest-benchmark==5.2.3
+pytest-repeat==0.9.4
 jsbeautifier==1.14.7
 
 datasets>=2.20,<3

